File: src/main/java/ua/softgroup/matrix/server/supervisor/producer/json/UserTimeAndCountResponse.java
package ua.softgroup.matrix.server.supervisor.producer.json;

/**
 * @author Oleksandr Tyshkovets <sg.olexander@gmail.com>
 */
public class UserTimeAndCountResponse extends TimeResponse {

    private long userId;
    private int totalAllSymbolsCount;
    private int windowsSwitchedCount;

    public UserTimeAndCountResponse() {
    }

    public UserTimeAndCountResponse(long userId, int workSeconds, int idleSeconds, double idlePercentage,
                                    int totalAllSymbolsCount, int windowsSwitchedCount) {
        this.userId = userId;
        this.totalWorkTimeSeconds = workSeconds;
        this.totalIdleTimeSeconds = idleSeconds;
        this.totalIdlePercentage = idlePercentage;
        this.totalAllSymbolsCount = totalAllSymbolsCount;
        this.windowsSwitchedCount = windowsSwitchedCount;
    }



    public long getUserId() {
        return userId;
    }

    public void setUserId(long userId) {
        this.userId = userId;
    }

    public int getTotalAllSymbolsCount() {
        return totalAllSymbolsCount;
    }

    public void setTotalAllSymbolsCount(int totalAllSymbolsCount) {
        this.totalAllSymbolsCount = totalAllSymbolsCount;
    }

    public int getWindowsSwitchedCount() {
        return windowsSwitchedCount;
    }

    public void setWindowsSwitchedCount(int windowsSwitchedCount) {
        this.windowsSwitchedCount = windowsSwitchedCount;
    }
}

File: src/main/scala/ua/softgroup/matrix/server/supervisor/producer/json/UserTimeAndCountResponse.scala
package ua.softgroup.matrix.server.supervisor.producer.json

import scala.beans.BeanProperty

/**
  * @author Oleksandr Tyshkovets <olexandr.tyshkovets@gmail.com>
  */
class UserTimeAndCountResponse extends TimeResponse {

  @BeanProperty
  var userId: Long = _

  @BeanProperty
  var totalAllSymbolsCount: Int = _

  @BeanProperty
  var windowsSwitchedCount: Int = _

  def this(userId: Long, workSeconds: Int, idleSeconds: Int, idlePercentage: Double,
           totalAllSymbolsCount: Int, windowsSwitchedCount: Int) {
    this()
    this.userId = userId
    this.totalWorkTimeSeconds = workSeconds
    this.totalIdleTimeSeconds = idleSeconds
    this.totalIdlePercentage = idlePercentage
    this.totalAllSymbolsCount = totalAllSymbolsCount
    this.windowsSwitchedCount = windowsSwitchedCount
  }

  override def toString: String = {
    "UserTimeAndCountResponse(userId=%d, totalWorkTimeSeconds=%d, totalIdleTimeSeconds=%d, totalIdlePercentage=%f, totalAllSymbolsCount=%d, windowsSwitchedCount=%d)"
      .format(userId, totalWorkTimeSeconds, totalIdleTimeSeconds, totalIdlePercentage, totalAllSymbolsCount, windowsSwitchedCount)
  }

}
